yarn-project/pxe/src/private_kernel/hints/compute_tx_include_by_timestamp.ts

```diff
@@ -0,0 +1,57 @@
+import { MAX_INCLUDE_BY_TIMESTAMP_DURATION } from '@aztec/constants';
+import type { PrivateKernelCircuitPublicInputs } from '@aztec/stdlib/kernel';
+import type { UInt64 } from '@aztec/stdlib/types';
+
+const ROUNDED_DURATIONS = [
+  3600, // 1 hour
+  1800, // 30 mins
+  1, // 1 second
+];
+
+function roundTimestamp(blockTimestamp: bigint, includeByTimestamp: bigint, duration: number): UInt64 {
+  const totalDuration = includeByTimestamp - blockTimestamp;
```

Comment:
I think we might want to round not only the `totalDuration` but the timestamp as a whole because if we round only the `totalDuration` then the tx carries with it an information of when it was constructed and the privacy set is a bit smaller. But I am not really sure if we really care about this because when a tx was constructed will be trackable by Aztec nodes (they can log when they received a given tx via p2p).

Curious if @iAmMichaelConnor has some opinion on this.

Reply:
I see. Didn't realize historical timestamp is public. It all makes sense then.

I guess in the future we might want to choose building against an older historical block if we would want to conceal this.

Thanks for the explanation ☺️